demos/text_detection_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,349 @@
+}
+
+bool ParseAndCheckCommandLine(int argc, char* argv[]) {
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+    if (FLAGS_i.empty()) {
+        throw std::logic_error("Parameter -i is not set");
+    }
+    if (FLAGS_m_td.empty() && FLAGS_m_tr.empty()) {
+        throw std::logic_error("Neither parameter -m_td nor -m_tr is not set");
+    }
+    if (!FLAGS_m_tr.empty() && FLAGS_dt.empty()) {
+        throw std::logic_error("Parameter -dt is not set");
+    }
+    return true;
+}
+
+int clip(int x, int maxVal) {
+    return std::min(std::max(x, 0), maxVal);
+}
+
+int main(int argc, char* argv[]) {
+    try {
+        // This demo covers certain topologies only
+        // Parsing and validating input arguments
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+        const bool gapiStreaming = !FLAGS_gapi_regular_mode;
+        const bool tdEnabled  = !FLAGS_m_td.empty();
+        const bool trEnabled  = !FLAGS_m_tr.empty();
+        const bool trComposite = std::string::npos != FLAGS_m_tr.find("encoder");
+        const float segmConfThreshold = static_cast<float>(FLAGS_cls_pixel_thr);
+        const float linkConfThreshold = static_cast<float>(FLAGS_link_pixel_thr);
+        const std::array<std::string,2> encoderOutputNames { FLAGS_out_enc_hidden_name,
+                                                             FLAGS_features_name        };
+        const std::array<std::string,3> decoderInputNames { FLAGS_in_dec_symbol_name,
+                                                            FLAGS_in_dec_hidden_name,
+                                                            FLAGS_features_name         };
+        const std::array<std::string,2> decoderOutputNames { FLAGS_out_dec_hidden_name,
+                                                             FLAGS_out_dec_symbol_name  };
+        if (FLAGS_pad.length() != 1) {
+            throw std::invalid_argument("Pad symbol should be 1 character");
+        }
+        const char kPadSymbol = FLAGS_pad[0];
+        std::string trSymbolsSet;
+        if (!tryReadVocabFile(FLAGS_m_tr_ss, trSymbolsSet)) {
+            trSymbolsSet = FLAGS_m_tr_ss;
+        }
+        if (trSymbolsSet.find(kPadSymbol) != trSymbolsSet.npos) {
+            throw std::invalid_argument("Symbols set for the Text Recongition model must not "
+                                        "contain the reserved symbol " + std::string{ kPadSymbol });
+        }
+
+        custom::NetsConfig config(FLAGS_m_td, FLAGS_m_tr);
+        if (tdEnabled) {
+            config.getTDinfo();
+            config.configureTD(FLAGS_d_td, FLAGS_w_td, FLAGS_h_td);
+            slog::info << "The Text Detection model " << FLAGS_m_td << " is loaded to " <<
+                FLAGS_d_td << slog::endl;
+        }
+        if (trEnabled) {
```

Comment:
I think you should put the body of this if and `if (tdEnabled)`'s body to `NetsConfig`'s constructor. Let the object manage itself.